packages/aws-cdk-lib/aws-ec2/README.md

```diff
@@ -810,7 +814,13 @@ If the security group ID is known and configuration details are unknown, use met
 const sg = ec2.SecurityGroup.fromLookupById(this, 'SecurityGroupLookup', 'sg-1234');
 ```
 
-The result of `SecurityGroup.fromLookupByName` and `SecurityGroup.fromLookupById` operations will be written to a file called `cdk.context.json`. You must commit this file to source control so that the lookup values are available in non-privileged environments such as CI build steps, and to ensure your template builds are repeatable.
+The result of `SecurityGroup.fromLookupByName` and `SecurityGroup.fromLookupById` operations will be
+written to a file called `cdk.context.json`. 
+You must commit this file to source control so that the lookup values are available in non-privileged
+environments such as CI build steps, and to ensure your template builds are repeatable.
+
+To customize the cache key use the `additionalCacheKey` parameter. 
+This can be useful if you want to scope the context variable to a construct (eg, using `additionalCacheKey: this.node.path`).
```

Comment:
Why would a user want to add the scope to the context variable? Why is this meaningfully different from creating multiple keys prefixed with the extra string?

Reply:
Why is this better than creating multiple context keys? Is it easier to track, easier to use, does it give a way to group related values together?

Reply:
Ah. If I understand your question correctly now, the upshot is that you otherwise can't! The context keys are determined by the construct (eg, when you call .fromLookup, it creates the key based on the lookup parameters), so this provides an escape hatch to customize them

Reply:
I'm not understanding the new capability you're proposing. Let's analyze an example. Today, if I want two values from lookup, I need two parameters, like this:

```
ssm.StringParameter.valueFromLookup(this, 'Parameter1');
ssm.StringParameter.valueFromLookup(this, 'Parameter2');
```

With your change, I have this now:

```
ssm.StringParameter.valueFromLookup(this, 'Parameter', 'extraKey1');
ssm.StringParameter.valueFromLookup(this, 'Parameter', 'extraKey2');
```

What does this do?

Reply:
This allows you to use the _same parameter_, but _separate the cache lifetimes_. The example I gave is with getting the latest Amazon Linux image - the name of the parameter is constant (AWS controls it!), and when spinning up new stateful instances I want the cache to include the value of the parameter at the time the instance was launched (which will be different for each instance).

Reply:
Ah! Now this is making sense, thank you for patiently explaining this. So when you pass the construct tree path (`this.node.path`) as the extra cache key and synthesize, the result of the parameter lookup is stored under `Parameter` as the top-level key, then `/path/to/my/instance` as the cache key. As you add more instances, you're adding another entry under `Parameter` to be `path/to/my/instance2`.

These all hold the same value. When the parameter changes, those previous context entries continue to retain that value. When you create new instances after the parameter changes though, you get another entry under `Parameter`, this time it's `path/to/my/instance3`, but it has a different value than the first two.

This is a really cool feature! Please update the READMEs to explain some of these details about the cache lifetimes. Also please give an example walkthrough, where you detail the value of a parameter, then show the context after doing a lookup with a certain lifetime key, and then show what happens when the parameter changes and you perform another lookup. This will help users understand how this feature works.

Reply:
Strictly speaking the path and the parameter name are concatenated to get the cache key, but you get it 🙂

I agree I should work on expanding the documentation, though I need to figure out how to balance descriptiveness and conciseness (I got what I interpreted as conflicting feedback on this previously)

Reply:
Please lean on the side of overly descriptive initially and then we can trim it down later, as needed.